File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int n = matrix.length;       
        int m = matrix[0].length;
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                if(matrix[i][j]==0){
                    for(int k=0;k<m;k++){
                        if(matrix[i][k]!=0){
                            matrix[i][k]=-545;
                        }
                    }
                    for(int k=0;k<n;k++){
                        if(matrix[k][j]!=0){
                            matrix[k][j]=-545;
                        }
                    }
                }
            }
        }
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                if(matrix[i][j]==-545){
                    matrix[i][j]=0;
                }
            }
        }
    }
    
}

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        
        n = len(matrix)          
        m = len(matrix[0])      

        for i in range(n):
            for j in range(m):

                if matrix[i][j] == 0:

                    # Mark row
                    for k in range(m):
                        if matrix[i][k] != 0:
                            matrix[i][k] = -545

                    # Mark column
                    for k in range(n):
                        if matrix[k][j] != 0:
                            matrix[k][j] = -545

        # Convert -1 to 0
        for i in range(n):
            for j in range(m):
                if matrix[i][j] == -545:
                    matrix[i][j] = 0    
        
        